saleor/graphql/shipping/mutations/shippings.py

```diff
@@ -202,13 +203,24 @@ def perform_mutation(cls, root, info, **data):
             info, data["shipping_method_id"], only_type=ShippingMethod
         )
         instances = []
-        for zip_range in data["input"]["zip_code_rules"]:
-            instance = models.ShippingMethodZipCodeRule.objects.create(
-                shipping_method=shipping_method,
-                start=zip_range["start"],
-                end=zip_range.get("end"),
-            )
-            instances.append(instance)
+        with transaction.atomic():
+            for zip_range in data["input"]["zip_code_rules"]:
+                try:
+                    start = zip_range["start"]
+                    end = zip_range["end"]
+                    instance = models.ShippingMethodZipCodeRule.objects.create(
+                        shipping_method=shipping_method, start=start, end=end,
+                    )
+                except IntegrityError:
+                    raise ValidationError(
+                        {
+                            "zipCodeRules": ValidationError(
+                                f"Entry start: {start}, end: {end} already exists.",
+                                code=ShippingErrorCode.ZIP_CODE_RULE_EXISTS.value,
+                            )
```

Comment:
Two questions:
- We already have `ShippingErrorCode.ALREADY_EXISTS`, can we use it instead of adding a new one?
- Don't we want to return to frontend any information about which list item from the zip code list already exists? Now we're returning just information that zipCodeRules field failed; some errors in our schema have additional information such as index of failing item.

Reply:
For both questions - yes, i'll add that 👍🏻